### PR TITLE
Suggest valid RethinkDB database name and prevent overwriting configuration settings

### LINKED ISSUE #174: RethinkDB configuration values overwritten when generating new service
The generator should look to see if RethinkDB connection settings exist in the configuration, and not prompt the user to re-enter a connection string.

### PR DESCRIPTION
**Pull request by luke3butler**
Prevent overwriting RethinkDB connection details by re-using existing
configuration settings.

Default to a valid RethinkDB database name.

closes #174 and #173

### CODE REVIEW

File: generators/connection/index.js [hunk 1]
```diff
@@ -243,6 +244,20 @@ module.exports = class ConnectionGenerator extends Generator {
             return false;
           }
 
+          if(typeof connectionString === 'object' && database === 'rethinkdb') {
```
Comment (daffl) [inline generators/connection/index.js]: Did this have to be added? It should already be done in https://github.com/feathersjs/generator-feathers/blob/master/generators/connection/index.js#L55

**Comment by daffl** [comment]
Weird that it doesn't like dashes. Why don't we just make the database name snake-cased in general?

File: generators/connection/index.js (hunk 1 above)
Comment (luke3butler) [inline generators/connection/index.js]: Nope, lines starting at 55 take care of creating a _new_ connection object, but that's useless during the prompts.

The `connectionString` pompt's `when` function checks to see if a connection is already defined in the configuration and skips the question if it exists. 
Right now it checks for a "string" and returns that string if it exists, but RethinkDB is an object.

My solution re-creates a connection string by destructuring the connection object, thus not tearing apart the way it currently works while keeping the already defined connection settings.

Does that make sense?

**Comment by luke3butler** [comment]
It looks like snake case is valid across each of the supported generator-feathers DBs. 
Is that the route we want to take?

**Comment by daffl** [comment]
@luke3butler In case you didn't see it, I released a new (pre) version of the generator which I think has all the issues you noticed fixed if you want to give it another try.

**Comment by ekryski** [comment]
I think snake case is fine. If that doesn't work an alternative would be camelcase.

**Comment by luke3butler** [comment]
@daffl Yep, everything I've spotted is working great.

**Pull request merged**